tools/caffe_translator/src/main/java/io/mxnet/caffetranslator/Optimizer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file Optimizer.java
+ * \brief Generates optimizer from solver prototxt
+ */
+
+package io.mxnet.caffetranslator;
+
+import org.stringtemplate.v4.ST;
+
+public class Optimizer {
+    private final GenerationHelper gh;
+    private final Solver solver;
+
+    public Optimizer(Solver solver) {
+        this.gh = new GenerationHelper();
+        this.solver = solver;
+    }
+
+    public String generateInitCode() {
+        ST st = gh.getTemplate("opt_" + solver.getType().toLowerCase());
+        if (st == null) {
+            System.err.println(String.format("Unknown optimizer type (%s). Using SGD instead.", solver.getType()));
+            st = gh.getTemplate("opt_sgd");
+        }
+
+        st.add("solver", solver);
```

Comment:
In case, st==null, optimizer is SGD. 
But solver.getType() will be still pointing to something other than SGD. Should we update solver here and then add st?
In case Optimizer St is used in other part.

Reply:
Solver is a representation of the solver prototxt file. If we update solver.type to SGD, other parts of the code won't be able to distinguish between (1) user decided solver type to be sgd and (2) we didn't understand the solver type the user is using. If we do that update, we will be loosing information and solver object will no more be an accurate representation of the solver prototxt. For now, there is no reason to do something like that.